### PR TITLE
feat: designed new features section

### PR DESCRIPTION
**Pull request by iryanraushan**
Screenshot from 2023-11-26 16-20-35
Screenshot from 2023-11-26 16-20-43
Screenshot from 2023-11-26 16-21-17
Screenshot from 2023-11-26 16-21-27

## Summary by CodeRabbit

- **New Features**
  - Revamped the Features section with a new grid layout and added images to feature cards for a more visual experience.
  - Renamed the Features component to Home, reflecting its broader role on the website.

- **Enhancements**
  - Updated the Home component to use Next.js's Image component for better performance and SEO.
  - Introduced unique IDs for feature items, enhancing accessibility and structure.

- **Style**
  - Enhanced the visual appeal of the feature cards with updated styling and layout adjustments.
  - Adopted a section-based layout for improved semantic structure and readability.

- **Refactor**
  - Transitioned to arrow function syntax for a more modern and concise codebase.
  - Extracted metadata and viewport configurations to their own exports for better modularity.

- **Documentation**
  - Removed outdated "and many more!" text to streamline content presentation.

### CODE REVIEW

**Comment by SaptarshiSarkar12** [comment]
@iryanraushan Please fix the website deployment error.
Error :point_down: 
```cpp
[16:26:53.998] Failed to compile.
[16:26:53.998] 
[16:26:53.998] ./app/Features.js
[16:26:53.999] 46:13  Error: Missing "key" prop for element in iterator  react/jsx-key
[16:26:53.999] 
[16:26:53.999] ./app/Header.js
[16:26:53.999] 93:6  Warning: React Hook useEffect has a missing dependency: 'onScroll'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps
[16:26:53.999] 
[16:26:53.999] info  - Need to disable some ESLint rules? Learn more here: https://nextjs.org/docs/basic-features/eslint#disabling-rules
[16:26:54.032] Error: Command "npm run build" exited with 1
```

**Review by SaptarshiSarkar12 — CHANGES_REQUESTED** [review]
@iryanraushan Please do not use saved images for the icons. Instead, you can do something like this :point_down: (Use **FontAwesome icons**)
For Open-Source icon, you can do
```javascript
icon: <div className={"text-6xl"}><i className={"fa-brands fa-osi fa-xl"} style={{ color: "#65a30d" }}></i></div>,
```
The size of the **OSI** image is the same as the SVG image of OSI that you have uploaded.
You can look for the icons [here](https://fontawesome.com/icons).

**Comment by SaptarshiSarkar12** [comment]
@iryanraushan Also, can the background color of the Features' section be changed? What if we have a blue shade gradient? Would that look nice :thinking:? Any other colours or gradients can be there - please let me know what you think we can have.

File: Website/app/Features.js [hunk 1]
```diff
@@ -1,60 +1,82 @@
 import React from "react";
-function Features() {
+import Image from "next/image";
+import Osi from "./img/osi.svg";
+import Download from "./img/download.svg";
+import YouTube from "./img/youtube.svg";
+import Desktop from "./img/desktop.svg";
+
+const Features = () => {
```
[the remark this answers is not in the document]
Comment (SaptarshiSarkar12) [reply Website/app/Features.js:8]: No, it will not be renamed.

File: Website/app/Features.js [hunk 2]
```diff
@@ -1,60 +1,82 @@
...
+                  {card.title}
+                </h2>
+
+                <div className="flex mt-2">
+                  <p className="text-sm leading-[24px] text-center">
+                    {card.description}
+                  </p>
+                  <Image
+                      src={card.icon}
+                      alt="card logo"
```
[the remark this answers is not in the document]
Comment (SaptarshiSarkar12) [reply Website/app/Features.js:67]: Will be looked into just before the merge.

File: Website/app/Features.js [hunk 3]
```diff
@@ -1,60 +1,82 @@
...
   return (
-      <div className="py-6 px-12 background bg-gradient-to-r from-cyan-500 to-blue-500 text-white font-sans select-none">
-        <h1 className="text-center mb-6 font-extrabold text-4xl pt-6">
-          Features
-        </h1>
-        <div className="column-1 flex flex-wrap justify-center gap-1">
-          {features.map((feature, index) => (
+      <section className="p-5">
+        <div className="text-center text-4xl font-semibold">Features</div>
+        <div className="grid grid-flow-row lg:grid-flow-col lg:[&>*:nth-of-type(1)]:row-span-2 lg:[&>*:nth-of-type(4)]:row-span-2 lg:items-center gap-6 grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 grid-rows-4 sm:grid-rows-2 mt-5  mx-auto;">
```
[the remark this answers is not in the document]
Comment (SaptarshiSarkar12) [reply Website/app/Features.js:47]: Will be done just before merge.

File: Website/app/Features.js [hunk 4]
```diff
@@ -1,60 +1,82 @@
...
-                <i
-                    className={feature.icon + " self-center mb-5 " + feature.color}
-                    aria-hidden="true"
-                    style={{ fontSize: "3rem" }}
-                ></i>
-                <h3 className="text-lg mb-1">{feature.title}</h3>
-                <p className="font-normal leading-normal">{feature.description}</p>
+                <div
+                    className="absolute w-full h-[3px] top-0 left-0"
+                    style={{ backgroundColor: `${card.color}` }}
```
[the remark this answers is not in the document]
Comment (SaptarshiSarkar12) [reply Website/app/Features.js:55]: Will be looked into just before merge.

**Comment by iryanraushan** [comment]
Which icons used ??

**Comment by SaptarshiSarkar12** [comment]
@iryanraushan You will replace each of the icon elements in `features` named constant, with the html tag I mentioned in the earlier comment.
You are supposed to use FontAwesome Icons.

**Comment by SaptarshiSarkar12** [comment]
@iryanraushan Did you understand what change I requested?

**Comment by iryanraushan** [comment]
i got it. But i can't use the icons which you used. I dont know why can u suggest me the full process because i tried so hard still no result. i also dont wanna use offline svg files. please sugget me what i can do ?

File: Website/app/Features.js [hunk 5]
```diff
@@ -1,60 +1,82 @@
...
+const Features = () => {
   const features = [
     {
+      id: 1,
       title: "It's Free and Open-Source",
       description:
           "Drifty is both free and open-source, so, you can bring new features that you wish to have 🎉",
-      icon: "fa-brands fa-osi",
-      color: "text-lime-600",
+      icon: Osi,
```
Comment (SaptarshiSarkar12) [inline Website/app/Features.js:15]:
```suggestion
      icon: <div className={"text-6xl"}><i className={"fa-brands fa-osi fa-xl"} style={{ color: "#65a30d" }}></i></div>,
```

File: Website/app/Features.js [hunk 6]
```diff
@@ -1,60 +1,82 @@
...
+      color: "#65a30d",
     },
     {
+      id: 2,
       title: "Faster Downloading of files",
       description:
           "Drifty supports parallel downloading using multiple threads",
-      icon: "fa fa-download",
-      color: "text-sky-400",
+      icon: Download,
```
Comment (SaptarshiSarkar12) [inline Website/app/Features.js:23]:
```suggestion
      icon: <div className={"text-6xl"}><i className={"fa-solid fa-download fa-xl"} style={{ color: "#38bdf8" }}></i></div>,
```

File: Website/app/Features.js [hunk 7]
```diff
@@ -1,60 +1,82 @@
...
+      color: "#38bdf8",
     },
     {
+      id: 3,
       title: "Support for downloading videos from YouTube, Instagram, etc.",
       description:
           "Drifty allows downloading videos from popular platforms like YouTube and Instagram with ease.",
-      icon: "fa fa-youtube-play",
-      color: "text-red-600",
+      icon: YouTube,
```
Comment (SaptarshiSarkar12) [inline Website/app/Features.js:31]:
```suggestion
      icon: <div className={"text-6xl"}><i className={"fa-brands fa-youtube fa-xl"} style={{ color: "#dc2626" }}></i></div>,
```

File: Website/app/Features.js [hunk 8]
```diff
@@ -1,60 +1,82 @@
...
+      color: "#dc2626",
     },
     {
+      id: 4,
       title: "Available both in GUI and CLI mode",
       description:
           "Drifty has both CLI and GUI mode. So, it can be used anywhere as CLI or as GUI according to the requirements of the user.",
-      icon: "fa fa-desktop",
-      color: "text-gray-700",
+      icon: Desktop,
```
Comment (SaptarshiSarkar12) [inline Website/app/Features.js:39]:
```suggestion
      icon: <div className={"text-6xl"}><i className={"fa-regular fa-display fa-xl"} style={{ color: "#374151" }}></i></div>,
```

File: Website/app/Features.js [hunk 9]
```diff
@@ -1,60 +1,82 @@
 import React from "react";
-function Features() {
+import Image from "next/image";
+import Osi from "./img/osi.svg";
+import Download from "./img/download.svg";
+import YouTube from "./img/youtube.svg";
+import Desktop from "./img/desktop.svg";
```
Comment (SaptarshiSarkar12) [inline Website/app/Features.js:6]: Remove the image imports
```suggestion
```

**Review by SaptarshiSarkar12 — CHANGES_REQUESTED** [review]
@iryanraushan Please commit the following requested changes and delete the `img` folder along with the images in that folder.

**Comment by iryanraushan** [comment]
Hey i want to ask something please help. i am new in open source but i feel good to do this type of project. please help me to setup this project on my local machine. 
all commands setup to un  ?

**Comment by SaptarshiSarkar12** [comment]
@iryanraushan Welcome to Open-Source :handshake:! I guess you have contributed to this project previously also.
Feeling good that you loved working on this project.
@iryanraushan Which part of the project do you want to setup? The **Drifty application** or the **Website**? I can help you out for sure :smile: :+1:.

**Pull request closed without merging**